lib/libcheri_caprevoke/libcheri_caprevoke.c

```diff
@@ -65,19 +65,19 @@ static uint64_t caprev_shadow_nomap_last_word_mask(ptraddr_t base, size_t len);
 static ptrdiff_t
 caprev_shadow_nomap_first_word_offset(ptraddr_t base)
 {
-	return (ptrdiff_t)(
+	return ((ptrdiff_t)(
 	    base / VM_CHERI_REVOKE_GSZ_MEM_NOMAP / 8 / sizeof(uint64_t)) *
-	    sizeof(uint64_t);
+	    sizeof(uint64_t));
 }
 
 static ptrdiff_t
 caprev_shadow_nomap_last_word_offset(ptraddr_t base, size_t len)
 {
 	if (len == 0)
-		return caprev_shadow_nomap_first_word_offset(base);
+		return (caprev_shadow_nomap_first_word_offset(base));
 
-	return (ptrdiff_t)((base + len - 1) / VM_CHERI_REVOKE_GSZ_MEM_NOMAP / 8
-	    / sizeof(uint64_t)) * sizeof(uint64_t);
+	return ((ptrdiff_t)((base + len - 1) / VM_CHERI_REVOKE_GSZ_MEM_NOMAP / 8
```

Comment:
Maybe move the `/` to the end of the preceding line while you are at it?

Reply:
Probably should also use rounddown2 rather than divide-then-multiply